pkg/api/util/util.go

```diff
@@ -8,52 +8,135 @@ package util
 
 import (
 	"crypto/subtle"
+	"crypto/tls"
+	"crypto/x509"
 	"fmt"
 	"net"
 	"net/http"
 	"strings"
 	"sync"
 
-	"github.com/DataDog/datadog-agent/pkg/api/security"
+	pkgtoken "github.com/DataDog/datadog-agent/pkg/api/security"
+	"github.com/DataDog/datadog-agent/pkg/api/security/cert"
 	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 var (
 	tokenLock sync.RWMutex
 	token     string
```

Comment:
I know it's a larger change that this PR is intending to fix, but it would be great if one day we could get rid of this global var. Perhaps instead this could be converted into a component?

Reply:
I completely agree ! The `authtoken.Component` already try to tackle this by wrapping it around.
The problem is that a lot of places in the code are relying on the auth_token, some in `pkg`.
The work involved in bringing the getter down to these points is quite big.